hooks/useUpdateTokenURI.ts

```diff
@@ -0,0 +1,57 @@
+import { useState } from "react";
+import { useTokenProvider } from "@/providers/TokenProvider";
+import { Address } from "viem";
+import useSignTransaction from "./useSignTransaction";
+import { getPublicClient } from "@/lib/viem/publicClient";
+import { CHAIN, CHAIN_ID } from "@/lib/consts";
+import { zoraCreator1155ImplABI } from "@zoralabs/protocol-deployments";
+import { useUserProvider } from "@/providers/UserProvider";
+import { uploadJson } from "@/lib/arweave/uploadJson";
+import { fetchTokenMetadata } from "@/lib/protocolSdk/ipfs/token-metadata";
+import getTokenInfo from "@/lib/viem/getTokenInfo";
+
+const useUpdateTokenURI = () => {
+  const { token, fetchTokenInfo } = useTokenProvider();
+  const { signTransaction } = useSignTransaction();
+  const { connectedAddress } = useUserProvider();
+
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const setMetadata = async (title: string, description: string) => {
```

Comment:
refactor the function name to match the hook and onchain function being called

```suggestion
  const updateTokenURI = async (title: string, description: string) => {
```